azure: switch flannel from VXLAN -> UDP

metral opened the pull request:
This is a work around on Azure to mitigate the VXLAN issues being seen ~as of kernel 4.11~: https://github.com/coreos/tectonic-installer/issues/1171

/cc @alexsomesan @s-urbaniak @euank @squat @justaugustus

metral commented:
I've tested this on `container_linux = "1409.7.0"` and it is working.

We should probably go ahead and also bump https://github.com/coreos/tectonic-installer/blob/master/config.tf#L74 to 1409.7.0 because the pinning of 1353.8.0 was a work-around for https://github.com/coreos/tectonic-installer/issues/1171.

Quentin-M commented:
ok to test

alexsomesan commented:
Azure tests passed. Seems there is an AWS flake.
Re-triggering.

alexsomesan commented:
retest this please

euank commented on modules/net/flannel-vxlan/resources/manifests/kube-flannel.yaml:
```diff
@@ -19,8 +19,8 @@ data:
       "Backend": {
-        "Type": "vxlan",
-        "Port": 4789
+        "Type": "udp",
```
This is still in a folder named `flannel-vxlan`; it should probably be moved/copied to be less confusing.

Also, does this folder really only impact azure?

metral commented:
Valid point about this change not only affecting Azure, but rather it
applies to all platforms. Hope catch @euank.

@*alexsomesan *thoughts on how to best limit this change to only Azure?

On Fri, Jul 28, 2017 at 9:58 AM Euan Kemp <notifications@github.com> wrote:

> *@euank* commented on this pull request.
> ------------------------------
>
> In modules/net/flannel-vxlan/resources/manifests/kube-flannel.yaml
> <https://github.com/coreos/tectonic-installer/pull/1526#discussion_r130136898>
> :
>
> > @@ -19,8 +19,8 @@ data:
>      {
>        "Network": "${cluster_cidr}",
>        "Backend": {
> -        "Type": "vxlan",
> -        "Port": 4789
> +        "Type": "udp",
>
> This is still in a folder named flannel-vxlan; it should probably be
> moved/copied to be less confusing.
>
> Also, does this folder really only impact azure?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/coreos/tectonic-installer/pull/1526#pullrequestreview-52984093>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABD6wCiQmuMPgkFN_rlStQN-rF53PKR5ks5sShMzgaJpZM4Ol_fL>
> .
>
-- 
Mike Metral
Senior Architect | Field Engineering
CoreOS

metral@coreos.com
305.282.7606

@mikemetral <https://twitter.com/mikemetral>

Quentin-M commented:
Could we simply swap vxlan by udp - without changing the port everywhere? If so, then you can make a variable for that, defaulting to vxlan, and override it in AZure only.

metral commented:
Currently testing UDP over port 4789, rather than default of 8285. Will report back and update PR per @Quentin-M's suggestion if successful.

Quentin-M commented:
Actually, I just saw that the port modification mostly happen in the Azure modules. So we could still do the port modification I guess. We'd have a condition over the protocol (vxlan/udp) for configuring the port, inside the flannel module.

Thing I noticed tho, the module is called 'flannel-vxlan'.. what was the rationale behind that naming? it would not make too much sense to have it configurable with that name?

metral commented:
I've modified the PR to:
- make the flannel backend type & port configurable with TFvars
- renamed the flannel-vxlan module to flannel
- bumped CL to 1409.7.0
- override the azure module to force the use of UDP and port 8285 with flannel

Everyone PTAL. Thanks.

alexsomesan commented on platforms/azure/tectonic.tf:
```diff
@@ -77,12 +77,14 @@ module "tectonic" {
-  cluster_cidr      = "${var.tectonic_cluster_cidr}"
+  flannel_image        = "${var.tectonic_container_images["flannel"]}"
+  flannel_cni_image    = "${var.tectonic_container_images["flannel_cni"]}"
+  flannel_backend_type = "udp"
```
May I suggest that we still pass the input variable to configure the type and in case of it not being set default to udp?

Something like:
`flannel_backend_type = "${var.tectonic_flannel_backend_type == "" ? "udp" : var.tectonic_flannel_backend_type}"`

metral replied:
By default, `flannel_backend_type` is being set to vxlan to ensure that all other providers other than Azure are set up appropriately per our recommended settings, and so that the user does not have to provide this variable themselves in their tfvars: https://github.com/coreos/tectonic-installer/pull/1526/files#diff-264131a8410d0f8f58af1c713d120c53R100

Therefore, `flannel_backend_type` won't ever be an empty string unless the user makes it so in their tfvars for Azure - an action we cannot rely on the user to do/know about, so I hardcoded it to `udp` for Azure to force it while we wait for a proper fix.

I'd like to to make the type be configurable with a default value available as you show in your suggestion, but I'm not sure if the current implementation and assumptions allow for it. Any ideas?

alexsomesan replied:
@metral Sorry, I missed the fact that we default the toplevel top `vxlan`. Your approach makes sense then, especially in the light of it being temporary while the vxlan issue is being addressed.

metral commented:
Closing in favor of https://github.com/coreos/tectonic-installer/pull/1526

squat commented:
@metral I think you mean #1556

metral commented:
Good catch @squat. It's late :)
